// search1api-mcp/build/resources.js
import { McpError, ErrorCode } from "@modelcontextprotocol/sdk/types.js";
import { log } from './utils.js';
/**
 * Define the default search query result resource
 */
export const DEFAULT_SEARCH_RESOURCE = {
    uri: "search1api://info",
    name: "Search1API Information",
    description: "Basic information about Search1API capabilities",
    mimeType: "application/json"
};
/**
 * List of all available resources
 */
export const RESOURCES = [DEFAULT_SEARCH_RESOURCE];
/**
 * Handle list resources request
 */
export function handleListResources() {
    log("Handling list resources request");
    return RESOURCES;
}
/**
 * Handle read resource request
 * @param resourceUri The resource URI to read
 */
export function handleReadResource(resourceUri) {
    log(`Handling read resource request for ${resourceUri}`);
    const resource = RESOURCES.find((r) => r.uri === resourceUri);
    if (!resource) {
        throw new McpError(ErrorCode.InvalidRequest, `Resource not found: ${resourceUri}`);
    }
    return resource;
}
